### PR TITLE
add binary_accuracy and validation_batch_size

### PR DESCRIPTION
**Pull request by ichuang**
This PR adds two small features, both needed for a variety of modeling tasks, such as wide+deep models.  The features are:

1. A `binary_accuracy` metric, which computes the average number of equal elements, between two binary-valued float tensors.  Contrast this with the existing accuracy metric, which does the same, but for one-hot encoded tensors.

2. A `validation_batch_size` parameter, used to specify a different batch size to be used for validation, versus for training.  Before, the same batch size was used for both.

Unit tests included.

### CODE REVIEW

**Comment by aymericdamien** [comment]
That looks great! Here are few comments:
1. Could you move this directly into 'accuracy' metric? Maybe check that, if labels are 1-D, then apply binary accuracy calculation. This way, 'accuracy' will work for both cases.
2. I am not sure why we need a validation_batch_size parameter? Sometimes for simplicity, it is better to not have too many parameters.
But it looks good overall! Thanks for contributing :)

**Comment by ichuang** [comment]
Re accuracy metric: good call, done & unit tests extended to cover.

Re `validation_batch`: right.  unfortunately, it's essential because otherwise, `batch_size` is used for validation, and this will cause metrics to be evaluated for the same batch size as is used for training.  This will in general give undesirably varying or wrong results for monitors, e.g. when computing confusion matrix entries with `validation_monitors`.  

An explicit relevant case is when the training set has 4096 entries, and the test set 256.  If `batch_size`=4096 is set, to use full-set batches for training, then this causes issues with the testing, which does not have that size available.

**Comment by aymericdamien** [comment]
@ichuang Right! Totally agree! Thanks for updating your code, I just tested it, everything seems to be working fine! I merge it :)
